### PR TITLE
nrunner: adding an external runner example (v2)

### LINKED ISSUE #5137: External Runner-like feature for nrunner
**Is your feature request related to a problem? Please describe.**
The legacy runner has a feature called external runner that allows users to add an executable that will actually run the test, taking the reference as an argument.

There are ways to do that with the nrunner architecture and the job API, but none that emulates the same behavior and command line arguments.

**Describe the solution you'd like**
Be able to use `--external-runner` command line and behavior with the `nrunner` architecture.

**Describe alternatives you've considered**
Not implement the same feature/command lines and let users rely on the job like like [here](https://github.com/avocado-framework/avocado/blob/2d68fe5f0d5dc3f78eaf1e5cf172b91670484d8e/examples/jobs/custom_exec_test.py#L17).

**Additional information**
The following [issue](https://github.com/avocado-framework/avocado/issues/4980) is related to this, as the external runner transform the test name.

### PR DESCRIPTION
**Pull request by beraldoleal**
This is a rough substitute for our external runner cases using the power
of our Job API.

Fixes #5137.

Signed-off-by: Beraldo Leal <bleal@redhat.com>
Signed-off-by: Cleber Rosa <crosa@rehdat.com>

#### Changes from v1:

 - Added documentation about external runner back
 - Added support to multiple references

### CODE REVIEW

File: docs/source/guides/user/chapters/advanced.rst [hunk 1]
```diff
@@ -82,6 +82,76 @@ And to select a different test runner, say, the legacy ``runner``::
...
+
+.. note:: This example is pretty obvious and could be achieved by giving
+   /tmp/pass and /tmp/fail shell “shebangs” (#!/bin/sh), making them executable
+   (chmod +x /tmp/pass /tmp/fail), and running them as “SIMPLE” tests.
+
+
+But now consider the following example::
+
+
+  $ $ avocado-external-runner curl redhat.com "google.com -v"
```
Comment (clebergnu) [inline docs/source/guides/user/chapters/advanced.rst:140]: Duplicate `$` here.

**Review by clebergnu — CHANGES_REQUESTED** [review]
Besides the minor documentation typo, LGTM.

**Review by ana — APPROVED** [review]
I made a test and it works exactly as the documentation says.
Except for the typo mentioned by @clebergnu this looks good.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**